rfcs/011-resource-access-control/011-resource-access-control.md

```diff
@@ -149,43 +341,41 @@ flowchart TD
-resources for the console such that only authenticated admins can access them. 
-
 ## Drawbacks
 
-> Why should we *not* do this? Please consider the impact on users,
-on the integration of this change with other existing and planned features etc.
+This solution proposes to introduce a new service: it will inherently increase the complexity of the project both in the
+development phase (coding, testing, releasing) as well as the operations (availability, scaling, performance).
 
-> There are tradeoffs to choosing any path, please attempt to identify them here.
+Additionally, the service will rely on the Postgres instance that we deploy alongside Renkulab as the backend, using a new
+database, thus increasing Renkulab's dependency on Postgres.
 
 ## Rationale and Alternatives
 
-> Why is this design the best in the space of possible designs?
+This design provides a robust and reliable interface for the Renkulab administrator to manage the user access to the
+resources and the resource definitions as well. It also provides a solid foundation to build upon more feature that
+concerns the Renkulab administrator, providing an interface that will evolve with time but should remain familiar.
+
+Other designs revolved around appending the information about users' access and resource pools to existing entities in
+Renkulab (ConfigMaps, Keycloak metadata etc...). Those designs have all the drawbacks of not providing an easy interface
+for the administrator to understand mistakes and issues with the configuration.
 
-> What other designs have been considered and what is the rationale for not choosing them?
+This feature is essential to reduce the spread of Renkulab.io instances that the SDSC maintains because many of those
+exist solely to dedicate compute resources to specific groups.
 
-> What is the impact of not doing this?
+Consolidating the Renkulab deployments has also the benefit of allowing data scientists from different groups to share
+their work across a single deployment.
 
 ## Unresolved questions
 
-> What parts of the design do you expect to resolve through the RFC process 
-before this gets merged?
+How to access the administrator interface and how it should look will have to be addressed in the context of
+discussing this RFC.
 
-> What parts of the design do you expect to resolve through the implementation 
-of this feature before stabilisation?
+The programming language chosen for the HTTP service and its dependencies will be chosen at the moment of implementation.
+At the time of implementation, some considerations on the required network policies so that the service can access Postgres
+will be needed.
 
-> What related issues do you consider out of scope for this RFC that could be 
-addressed in the future independently of the solution that comes out of this RFC?
+The problem addressed by this RFC is somewhat related to the topic of cross-cluster resource sharing.
```

Comment:
Generally speaking it is possible for any Kubernetes cluster: when you setup `kubelet` it only wants to know where to reach the controlplane, it does not make a distinction between local network or otherwise.

It has of course several caveats, for once the kubelet and the controlplane must be reachable in both ways, then there might be some latency issue and finally also some security concerns. Using a VPN could address many of the previous concerns.

This is also what I think about when speaking of cross-cluster resource sharing, at least in its simplest implementation.

In any case this is meant to be just a vague direction, I don't claim that I have the full design proposal for that at this moment :grin:

Reply:
Ok cool just wanted to make sure we are on the same page. Sounds good.